core/trino-main/src/main/java/io/trino/execution/SetViewAuthorizationTask.java

```diff
@@ -80,16 +110,115 @@ public String getName()
+
+                case ROLE:
+                    checkCanSetAuthorizationToRole(session, metadata, accessControl, statement, viewName, principal);
+                    break;
+            }
+
+            checkCanAccessViewEntities(session, stateMachine, parameters, viewName, view, principal);
+            checkTargetPrincipalCanCreateView(session, transactionManager, accessControl, viewName, principal);
         }
+
         accessControl.checkCanSetViewAuthorization(session.toSecurityContext(), viewName, principal);
 
         metadata.setViewAuthorization(session, viewName.asCatalogSchemaTableName(), principal);
 
         return immediateVoidFuture();
     }
+
+    private void checkCanSetAuthorizationToUser(QualifiedObjectName viewName, TrinoPrincipal principal)
+    {
+        // TODO: implement a separate permission to transfer ownership to another user
+        denySetViewAuthorization(viewName.toString(), principal);
+    }
+
+    private void checkCanSetAuthorizationToRole(Session session, Metadata metadata, AccessControl accessControl, SetViewAuthorization statement, QualifiedObjectName viewName, TrinoPrincipal principal)
+    {
+        // The logic for ROLEs is that the user should be able to transfer ownership to any role they can assume,
+        // as this does not give them any additional privileges they can't already obtain:
+        Optional<String> catalog = processRoleCommandCatalog(metadata, session, statement, Optional.of(viewName.getCatalogName()));
+        if (catalog.isPresent()) {
+            try {
+                accessControl.checkCanSetCatalogRole(SecurityContext.of(session), principal.getName(), catalog.get());
+            }
+            catch (AccessDeniedException e) {
+                wrapAccessDeniedException(viewName, principal, e);
+            }
+        }
+        else {
+            Set<RoleGrant> roleGrants = metadata.listApplicableRoles(session, new TrinoPrincipal(USER, session.getUser()), Optional.empty());
+            if (roleGrants.stream().map(RoleGrant::getRoleName).noneMatch(principal.getName()::equals)) {
+                denySetViewAuthorization(viewName.toString(), principal, "Not a member of role " + principal.getName());
+            }
+        }
+    }
+
+    private void checkCanAccessViewEntities(Session session, QueryStateMachine stateMachine, List<Expression> parameters, QualifiedObjectName viewName, ViewDefinition view, TrinoPrincipal targetPrincipal)
+    {
+        CreateView statement = new CreateView(
+                QualifiedName.of(viewName.getCatalogName(), viewName.getSchemaName(), viewName.getObjectName()),
+                (Query) sqlParser.createStatement(view.getOriginalSql(), createParsingOptions(session)),
+                false,
+                view.getComment(),
+                Optional.of(DEFINER));
+        try {
+            analyzerFactory.createAnalyzer(session, parameters, parameterExtractor(statement, parameters), stateMachine.getWarningCollector())
```

Comment:
don't you need to use `targetPrincipal` as session identity? Because you verify if current identity can create a view and access view entities, not the one who will be the "owner"